src/state/action-creators.js

```diff
@@ -291,7 +291,9 @@ export const drop = () =>
 
     if (state.phase !== 'DRAGGING') {
       console.error('cannot drop if not dragging', state);
-      dispatch(clean());
+      // We need to wrap this in a setTimeout to avoid a race
+      // condition with `lift`, which includes timeouts
+      setTimeout(() => dispatch(clean()));
```

Comment:
This is the actual fix. The bug was occurring when a drop was triggered before a lift was completed. The `lift` action includes two `setTimeout` calls and ends up putting the app in the `DRAGGING` state. If a drop is triggered before the timeout callback firest the app ends up stuck in the `DRAGGING` state even though the item has already been dropped. The simplest way to fix this is to wrap the `clean()` action here in a timeout of its own. This means it will fire after the `lift` finishes and will ensure that the app is always put back in the `IDLE` state.

Reply:
nice

Reply:
Seems reasonable for now. Do you think there is any additional checks we should do here?

Reply:
the most robust thing to do would be to store a reference to the `setTimeout` somewhere so that we can clear it, but I think this is a much less complex solution. fwiw we won't even need that timeout if this issue gets resolved: https://github.com/chenglou/react-motion/issues/437